linode/rancher-ui-driver-linode
# Bump default image to Ubuntu 22.04 and default region to us-iad

## 📝 Description

This change updates the default image to `Ubuntu 22.04 LTS` and the default region to `us-iad` when creating Linode/RKE1 node templates. This is necessary because the previous default (`Ubuntu 18.04 LTS`) is outdated and the previous region (`us-east`) is a legacy compute region that we are looking to move away from.

Once this PR has been merged we should cut a release and put up a corresponding PR to bump the driver upstream.

See https://github.com/rancher/rancher/issues/44179

## ✔️ How to Test

The following manual testing steps assume you have provisioned a Rancher instance using Docker (named `rancher`) and have cloned this repository onto a development Linode.

1. Enter the project directory:

```bash
cd rancher-ui-driver-linode
```

2. Build the UI driver:

```bash
npm run build
```

3. Copy the built UI driver into the running Rancher instance:

```bash
docker cp ./dist/. rancher:/usr/share/rancher/ui/assets/rancher-ui-driver-linode
```

4. Navigate to the RKE1/Linode cluster creation in the web UI of your Rancher instance:

```
https://my.rancher.instance/dashboard/c/_/manager/provisioning.cattle.io.cluster/create?type=linode
```

5. Attempt to add a node template.
6. Ensure the default selected image is `Ubuntu 22.04 LTS` and the default selected region is `US-Mia (US)`.
7. Provision a cluster with these defaults and ensure the cluster becomes ready.

## 📷 Preview

<img width="1051" alt="Rancher" src="https://github.com/linode/rancher-ui-driver-linode/assets/114949949/cae2d1e6-2be3-401b-af6c-e9850d1f9469">
